src/osd/OSD.h

```diff
@@ -1610,6 +1614,12 @@ class OSD : public Dispatcher,
 	p->complete(0);
       }
     }
+
+    double get_cost_per_io() const {
+      uint32_t shard_index = 0;
+      auto &&sdata = osd->shards[shard_index];
```

Comment:
auto &&sdata isn't necessary -- use auto &sdata.  Remove the shard_index variable entirely and replace it with 0.